supervisor.go

```diff
@@ -353,6 +353,28 @@ func (s *Supervisor) Getpid(ctx context.Context, _ struct{}, pid *int) error {
 	return nil
 }
 
+type shutdownRequest struct {
+	Delay   time.Duration
+	Message string
+}
+
+// Shutdown will cause the process to exit aysnchronously at a point
+// in the future no sooner than the specified delay.
+func (s *Supervisor) Shutdown(ctx context.Context, req shutdownRequest, _ *struct{}) error {
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		wg.Done()
+		time.Sleep(req.Delay)
+		log.Print(req.Message, "\n")
+		os.Exit(0)
```

Comment:
this should call `s.system.Exit(1)`.

(e.g., in-process implementations do different things.)